src/docs/sdk/hub_and_scope_refactoring.mdx

```diff
@@ -170,14 +172,84 @@ For more information like the goals (and non-goals) see this GitHub Issue:
+
+
+#### Hook
+
+In order to fork our scopes, when required by OTel, we need some sort of a hook. We're not yet completely sure, when forking is actually required. In theory it should suffice to fork scopes, whenever a new OTel span is created, or maybe even only when the span is saved. What hooks are available in OTel depends on the language, so this will likely differer from SDK to SDK.
+
+- Context Forking: Whenever a modified copy of an OTel `Context` is created using `context.with(...)`.
+- Context Saving: Whenever an OTel `Context` is stored using `context.makeCurrent()`
+- Span Creation: Whenever a new OTel span is created, regardless of it being stored in any `Context` or that `Context` being stored.
+
+For `Context` forking / storing we could filter changes to only fork scopes when the span changes in order to have fewer forked scopes.
+
+If no hook is available, we have to fall back to a more complex variant of global storage. See the middle example in [this miro board](https://miro.com/app/board/uXjVNnI7dsE=/).
+
+A `Context` fork/storage hook alone unfortunately does not allow us to do everything we need. In `SpanProcessor.onEnd` and by extension also `SpanExporter.export`, `Context.current()` returns the parent Context and not the `Context` that was active during span execution. This seems to be by design in OTEL. This means even if we managed to add our Scopes to `Context`, we can't access it in `SpanProcessor.onEnd()`.
+
+#### Storing Scopes which Belong to a Certain Span
+
+If your language allows you to modify the OTel span and add Sentry `Scopes` to the span, that's probably the easiest way of getting scopes into `SpanProcessor.onEnd` and by extension `SpanExporter.export`. However we haven't explored this variant in detail yet.
+
+If modifying the span isn't possible, the only way to retrieve the `Scopes` in `SpanExporter.export`, we've found so far, is to use a global storage. For this global storage, we use OTel spans as weakly referenced keys and Sentry `Scopes` as values. We use weak references, so we don't cause memory leaks, e.g. in case a span is never finished.
+
+The hook for OTel span creation / `Context` forking / storing then needs to look up Sentry `Scopes` in either the span (if the language supports this variant) or the global storage.
+
+Scopes forked in `SpanProcessor.onStart` can be found via the OTel span in `Context` (or global storage if span can't be modified). However, there may already be a `Scopes` object on the `Context`. The hook has to figure out which `Scopes` object to fork. An ancestry check can be used to see if the `Scopes` from `Context` is an ancesor of the `Scopes` of the span (or from global storage). If so, we fork the `Scopes` from the span (or global storage), since it's the "youngest". If that's not the case, we fork `Scopes` from `Context` or fall back to forking root scopes, if there's no `Scopes` on the `Context`.
```

Comment:
```suggestion
Scopes forked in `SpanProcessor.onStart` can be found via the OTel span in `Context` (or global storage if span can't be modified). However, there may already be a `Scopes` object on the `Context`. The hook has to figure out which `Scopes` object to fork. An ancestry check can be used to see if the `Scopes` from `Context` is an ancestor of the `Scopes` of the span (or from global storage). If so, we fork the `Scopes` from the span (or global storage), since it's the "youngest". If that's not the case, we fork `Scopes` from `Context` or fall back to forking root scopes, if there's no `Scopes` on the `Context`.
```